### ISSUE TITLE
Robot state publisher log large amount of "moved backward in time" warnings.

### ISSUE THREAD

**Issue by mqt0029** [usnistgov/ARIAC#310]
On both containerized environment and direct ROS 2 Iron installation on a fresh Ubuntu 22.04 Jammy installation, when launching ARIAC with `ros2 launch ariac_gazebo ariac.launch.py`, I am seeing a constant stream of warnings:
```
[robot_state_publisher-7] [WARN] [1708121919.739176014] [robot_state_publisher]: Moved backwards in time, re-publishing joint transforms!
[robot_state_publisher-7] [WARN] [1708121919.840994544] [robot_state_publisher]: Moved backwards in time, re-publishing joint transforms!
[robot_state_publisher-7] [WARN] [1708121919.939366885] [robot_state_publisher]: Moved backwards in time, re-publishing joint transforms!
[robot_state_publisher-7] [WARN] [1708121920.039513745] [robot_state_publisher]: Moved backwards in time, re-publishing joint transforms!
```

and about 5% of the runs, the logs returns this, which might be related:

```
[environment_startup_node.py-5] Warning: TF_OLD_DATA ignoring data from the past for frame floor_wrist_2_link at time 5.804000 according to authority default_authority
[environment_startup_node.py-5] Possible reasons are listed at http://wiki.ros.org/tf/Errors%20explained
[environment_startup_node.py-5]          at line 292 in ./src/buffer_core.cpp
[environment_startup_node.py-5] Warning: TF_OLD_DATA ignoring data from the past for frame floor_wrist_3_link at time 5.804000 according to authority default_authority
[environment_startup_node.py-5] Possible reasons are listed at http://wiki.ros.org/tf/Errors%20explained
[environment_startup_node.py-5]          at line 292 in ./src/buffer_core.cpp
[environment_startup_node.py-5] Warning: TF_OLD_DATA ignoring data from the past for frame torso_main at time 5.804000 according to authority default_authority
[environment_startup_node.py-5] Possible reasons are listed at http://wiki.ros.org/tf/Errors%20explained
[environment_startup_node.py-5]          at line 292 in ./src/buffer_core.cpp
[environment_startup_node.py-5] Warning: TF_OLD_DATA ignoring data from the past for frame torso_tray at time 5.804000 according to authority default_authority
[environment_startup_node.py-5] Possible reasons are listed at http://wiki.ros.org/tf/Errors%20explained
[environment_startup_node.py-5]          at line 292 in ./src/buffer_core.cpp
[environment_startup_node.py-5] Warning: TF_OLD_DATA ignoring data from the past for frame small_rail at time 5.804000 according to authority default_authority
[environment_startup_node.py-5] Possible reasons are listed at http://wiki.ros.org/tf/Errors%20explained
[environment_startup_node.py-5]          at line 292 in ./src/buffer_core.cpp
[environment_startup_node.py-5] Warning: TF_OLD_DATA ignoring data from the past for frame torso_base at time 5.804000 according to authority default_authority
[environment_startup_node.py-5] Possible reasons are listed at http://wiki.ros.org/tf/Errors%20explained
[environment_startup_node.py-5]          at line 292 in ./src/buffer_core.cpp
[environment_startup_node.py-5] Warning: TF_OLD_DATA ignoring data from the past for frame robot_base at time 5.804000 according to authority default_authority
[environment_startup_node.py-5] Possible reasons are listed at http://wiki.ros.org/tf/Errors%20explained
[environment_startup_node.py-5]          at line 292 in ./src/buffer_core.cpp
```

Does ARIAC dev team have any pointer as to why this might be happening? When this happens, either one of the robots or both starts to droops and does not responds to control.

I have also attached a full launch log for context. Weirdly enough, this does not happen on Ubuntu 22.04 on Windows 11 via WSL2.

> :spiral_notepad: [**launch_log.txt**](https://github.com/usnistgov/ARIAC/files/14316534/launch_log.txt)

**Comment by jaybrecht** [comment]
My best guess is that there is another node on your system that is publishing TF data/joint state data that was not properly exited. Try running

`ros2 daemon stop`

then

`ros2 daemon start`

then restarting the competition.

**Comment by mqt0029** [comment]
Thanks for the suggestion, but that wasn't it.

As it turns out, I copy-pasted an earlier `docker run` snippet from old code, which has the flag `--network host` on it. While sharing the docker image between the team members on the same LAN, all running Ubuntu, we have a couple of instances of ARIAC on various machines, all has `--network host` in it. Removing the flag solved the problem as the containers no longer sees other ARIAC instances topics. The simulator started right up, with no errors.

Docker on WSL2 ignored the `host` option, hence it worked there from the get go.

**TL;DR: Be careful with Docker's `--network host` flag!**
